sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -84,7 +85,7 @@ class OperationFailed(Exception):
     pass
 
 
-def _as_json(response: "ResponseType") -> Dict[str, Any]:
+def _as_json(response: "HTTPResponseType") -> Dict[str, Any]:
```

Comment:
maybe for another PR but now that we have updated mypy, which was support for recursive type aliases maybe we can add in a JSON type as described [here](https://github.com/python/typing/issues/182). It'll be much nicer to read `JSON` as the return :)